lib/auth.js

```diff
@@ -142,6 +142,33 @@ bedrock.events.on('bedrock-session-http.session.get', function(req, session) {
 
 bedrock.events.on('bedrock-views.vars.get', addViewVars);
 
+bedrock.events.on('bedrock-identity.insert', function(options, callback) {
+  // add IDP identity defaults
+  var identity = options.identity;
+
+  var defaults = bedrock.util.clone(config.idp.defaults);
+  var clonedIdentity = bedrock.util.extend(
+    {}, defaults.identity, bedrock.util.clone(identity));
+  // Must mutate original identity object, so assign all props back over
+  for(var key in clonedIdentity) {
+    identity[key] = clonedIdentity[key];
+  }
+  // create identity ID from slug if not present
+  if(!('id' in identity)) {
+    identity.id = api.createIdentityId(identity.sysSlug);
+  }
+
+  // if using a 'did:', add resource role for local id
+  if(identity.id.indexOf('did:') === 0) {
+    identity.sysResourceRole = identity.sysResourceRole || [];
+    identity.sysResourceRole.push({
+      sysRole: 'identity.registered',
+      resource: [api.createIdentityId(identity.sysSlug)]
```

Comment:
Sounds good. The semantics for this changed a bit because we previously had to make an explicit call to run this code -- and now anyone who inserts an identity that has the idp installed will have this code run, so we need to make sure we check appropriate flags or whatever.